cmd/clusterctl/cmd/plugin_test.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+var baseArgs = []string{"clusterctl", "foo"}
+
+const (
+	forkEnvVar          = "CLUSTERCTL_PLUGIN_TEST_FORK"
+	pluginCommandEnvVar = "CLUSTERCTL_PLUGIN_COMMAND"
+)
+
+func Test_plugin(t *testing.T) {
+	// If CLUSTERCTL_PLUGIN_TEST_FORK is set then just execute the command.
```

Comment:
It's highly confusing that this test calls itself via exec. Is there any other way to achieve what you want to achieve?

Reply:
Yes, I understand your confusion, but unfortunately there are no other unit testing options for this case, since the `Execute` function includes a `syscall.Exec`, which does not return control and cannot be tested after exiting.

In this case, only real e2e test based on build `clusterctl` binary, but not smoke unit-test.